app/service/rest.py

```diff
@@ -312,6 +318,29 @@ def update_whitelist(service_id):
         return '', 204
 
 
+@service_blueprint.route('/<uuid:service_id>/deactivate', methods=['POST'])
+def deactivate_service(service_id):
+    service = dao_fetch_service_by_id(service_id)
+
```

Comment:
That would not solve the problem. We want this action to be all or nothing. I think you need to create one function to do all updates and commit at the end (or rollback)